src/libbluechi/common/time-util.c

```diff
@@ -42,6 +34,23 @@ double micros_to_millis(uint64_t time_micros) {
         return (double) time_micros * microsec_to_millisec_multiplier;
 }
 
+uint64_t timespec_to_micros(struct timespec time) {
+        uint64_t sec_micros = 0;
+        uint64_t nsec_micros = 0;
+
+        if ((uint64_t) time.tv_sec > UINT64_MAX / sec_to_microsec_multiplier) {
```

Comment:
Not sure if these checks for the maximum are really necessary since the actual times will likely be much lower than what `uint64_t` can store. 
What do you think?

Reply:
I think overflow checking is a good habit to have to prepare for unusual situations. There is already similar code in systemd.